### PR TITLE
Import Circle Game (used for testsuite)

### PR DESCRIPTION
**Pull request by jdetter**
## Description of Changes
*Describe what has been changed, any new features or bug fixes*

- I have taken the circle game and imported it 
- This is the base PR for this: https://github.com/clockworklabs/com.clockworklabs.spacetimedbsdk/pull/176

## API

 - [ ] This is an API breaking change to the SDK

*If the API is breaking, please state below what will break*

## Requires SpacetimeDB PRs
*List any PRs here that are required for this SDK change to work*

None

## Testing
*Write instructions for a test that you performed for this PR*

- [x] dotnet testsuite still passes
- [x] testsuite builds, although there are no tests to run because those are in a separate PR (https://github.com/clockworklabs/com.clockworklabs.spacetimedbsdk/pull/176)

proof this builds:

image

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -75,3 +75,4 @@ obj~
 # This is used for local paths to SpacetimeDB packages.
 /nuget.config
 /nuget.config.meta
+.idea/
```
Comment (jdetter) [inline .gitignore:78]: unrelated change: This ignores the jetbrains `.idea/` directory which is created when opening the project in rider.

File: SpacetimeDB.ClientSDK.csproj [hunk 2]
```diff
@@ -19,19 +19,31 @@
...
 
+  <!-- Include non-code files -->
   <ItemGroup>
     <None Update="logo.png" Pack="true" PackagePath="" />
     <None Update="README.md" Pack="true" PackagePath="" />
   </ItemGroup>
 
+  <!-- Explicitly include all .cs files except Unity-related scripts -->
+  <ItemGroup>
+    <Compile Include="src/**/*.cs" Exclude="unity-tests/client/Assets/**/*.cs" />
```
Comment (jdetter) [inline SpacetimeDB.ClientSDK.csproj:43]: This prevents the unity testsuite cs files from being compiled during the dotnet testsuite.

File: .npmignore [hunk 3]
```diff
@@ -0,0 +1 @@
+unity-tests/
```
Comment (jdetter) [inline .npmignore]: this prevents the `unity-tests` directory from being downloaded when a user imports our SDK

Comment (RReverser) [reply .npmignore]: I don't think .npmignore is respected for Git deps? IIUC Unity just uses Git clone, which doesn't have a way to exclude individual folders. We can only use the `~` suffix to exclude files and folders from assets, which you're already doing after the latest commits.

**Comment by cloutiertyler** [comment]
This comment also applies to this precursor PR: https://github.com/clockworklabs/com.clockworklabs.spacetimedbsdk/pull/176#issuecomment-2444969797

**Comment by cloutiertyler** [comment]
I am going to evaluate whether this should be a submodule prior to merging this.

**Comment by Centril** [comment]
Rebased this atop of staging (trivial conflict to resolve, rc1 instead of 0.12.*)

**Comment by jdetter** [comment]
closed in favor of using this as a submodule https://github.com/clockworklabs/com.clockworklabs.spacetimedbsdk/pull/186
